librz/core/project.c

```diff
@@ -38,22 +42,43 @@ RZ_API RzProjectErr rz_project_save(RzCore *core, RzProject *prj, const char *fi
 }
 
 RZ_API RzProjectErr rz_project_save_file(RzCore *core, const char *file) {
+	char *tmp_file;
+	int mkstemp_fd = rz_file_mkstemp("svprj", &tmp_file);
+
+	if (mkstemp_fd == -1 || !tmp_file) {
+		close(mkstemp_fd);
```

Comment:
if you have to do this both in the if and outside, just put it before the `if` one time.